### PR TITLE
Made ds_EffectHookScripts, added enum to oEffectManager and replaced …

### PR DESCRIPTION
**Pull request by nick-wilson95**
…map_effect_to_bucket_handler with new code

### CODE REVIEW

**Comment by nick-wilson95** [comment]
[Updates to effect data](https://trello.com/c/pi2Bpon6/5-updates-to-effect-data)

File: scripts/ds_EffectHookScripts_Destroy/ds_EffectHookScripts_Destroy.gml [hunk 1]
```diff
@@ -0,0 +1,15 @@
+/// @function ds_EffectHookScripts_Destroy(ds_EffectHookScripts_data) Destroys the given ds_EffectHookScripts.
+/// @param {ds_EffectHookScripts} ds_EffectHookScripts_data The ds_EffectHookScripts to destroy.
+/// @returns {boolean} True if the EffectHookScripts was destroyed.
+var _ds_EffectHookScripts_data = argument[0];
+
+
+if (!ds_EffectHookScripts_InstanceOf(_ds_EffectHookScripts_data)) {
+	ErrorHandler_Error(
+		"Variable was not destroyed as it was of type " + typeof(_ds_EffectHookScripts_data) + " and not of type ds_EffectHookScripts"
```
Comment (Jibzi) [inline scripts/ds_EffectHookScripts_Destroy/ds_EffectHookScripts_Destroy.gml]: Split this string onto new lines at each +.

File: scripts/ds_EffectHookScripts_GetScript/ds_EffectHookScripts_GetScript.gml [hunk 2]
```diff
@@ -0,0 +1,21 @@
...
+/// @param {ds_EffectHookScripts} ds_EffectHookScripts_data The ds_EffectHookScripts to get the script for.
+/// @param {enum} enum_eventHook The hook to get the script for.
+/// @returns {scr} The script if one exists, undefined otherwise.
+var _ds_EffectHookScripts_data = argument[0];
+var _enum_eventHook = argument[1];
+
+
+if (!ds_EffectHookScripts_InstanceOf(_ds_EffectHookScripts_data)) {
+	ErrorHandler_Error(
+		"Could not get script for input, as it was of type " + typeof(_ds_EffectHookScripts_data) + " and not of type ds_EffectHookScripts"
```
Comment (Jibzi) [inline scripts/ds_EffectHookScripts_GetScript/ds_EffectHookScripts_GetScript.gml]: Same as above comment.

File: scripts/ds_EffectHookScripts_InstanceOf/ds_EffectHookScripts_InstanceOf.gml [hunk 3]
```diff
@@ -0,0 +1,25 @@
...
+	_any_currentKey = ds_map_find_first(_any_data);	
+}
+
+for(var _num_key = 0; _num_key < ds_map_size(_any_data); _num_key++) {
+	if(!is_real(_any_currentKey) or !script_exists(_any_data[? _any_currentKey])){
+		return false;
+	}
+	
+	_any_currentKey = ds_map_find_next(_any_data, _any_currentKey);
+	if(_any_currentKey == undefined) break;
```
Comment (Jibzi) [inline scripts/ds_EffectHookScripts_InstanceOf/ds_EffectHookScripts_InstanceOf.gml]: This break will then result in a return of true, which I'm not sure is correct.

File: scripts/ds_EffectHookScripts_ScriptExists/ds_EffectHookScripts_ScriptExists.gml [hunk 4]
```diff
@@ -0,0 +1,17 @@
...
+/// @param {ds_EffectHookScripts} ds_EffectHookScripts_data The ds_EffectHookScripts to check on.
+/// @param {enum} enum_eventHook The hook to check for.
+/// @returns {bool} True if the script exists.
+var _ds_EffectHookScripts_data = argument[0];
+var _enum_eventHook = argument[1];
+
+
+if (!ds_EffectHookScripts_InstanceOf(_ds_EffectHookScripts_data)) {
+	ErrorHandler_Error(
+		"Could not check for script existence, as input was of type " + typeof(_ds_EffectHookScripts_data) + " and not of type ds_EffectHookScripts"
```
Comment (Jibzi) [inline scripts/ds_EffectHookScripts_ScriptExists/ds_EffectHookScripts_ScriptExists.gml]: Same again.

File: scripts/ds_EffectHookScripts_SetScript/ds_EffectHookScripts_SetScript.gml [hunk 5]
```diff
@@ -0,0 +1,27 @@
...
+/// @param {scr} scr_script The script to set to.
+/// @returns {bool} True if the script was set successfully.
+var _ds_EffectHookScripts_data = argument[0];
+var _enum_eventHook = argument[1];
+var _scr_script = argument[2];
+
+
+if (!ds_EffectHookScripts_InstanceOf(_ds_EffectHookScripts_data)) {
+	ErrorHandler_Error(
+		"Could not set script for input, as it was of type " + typeof(_ds_EffectHookScripts_data) + " and not of type ds_EffectHookScripts"
```
Comment (Jibzi) [inline scripts/ds_EffectHookScripts_SetScript/ds_EffectHookScripts_SetScript.gml]: One more time.

**Pull request merged**